### PR TITLE
TES node runner project

### PR DESCRIPTION
**Pull request by giventocode**
This is the initial PR of the TES node runner.  A few things to call out:

 - This PR includes three new projects, `Tes.Runner` class library with the core implementation, `Tes.Runner.Test` test project for the library, and `Tes.RunnerCLI` a lightweight CLI wrapper for the library and the entry-point for execution.  Checkout [README](https://github.com/microsoft/ga4gh-tes/blob/ja-tes-runner/Tes.RunnerCLI/README.md) for more details.
 - The initial focus is on the download and upload operations so we can migrate those first - checkout  #148 for more details.
 - Contains initial implementation for the Docker execution using the Docker API,  but follow up work is needed to iron out edge-cases and better error handling. 
 - The test project, includes integration tests, however these are disabled (`[Ignore]`) for now as they require `Azurite` which I think is not included in our build pipelines.

### CODE REVIEW

File: Tes.Runner/Executor.cs [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        {
+            logger.LogInformation($"Writers:{blobPipelineOptions.NumberOfWriters}");
+            logger.LogInformation($"Readers:{blobPipelineOptions.NumberOfReaders}");
+            logger.LogInformation($"Capacity:{blobPipelineOptions.ReadWriteBuffersCapacity}");
+            logger.LogInformation($"BlockSize:{blobPipelineOptions.BlockSize}");
+        }
+
+        private static async Task<TimeExecutionResult<T>> TimedExecutionAsync<T>(Func<Task<T>> execution)
+        {
+            var sw = new Stopwatch();
```
Comment (MattMcL4475) [inline Tes.Runner/Executor.cs:143]: Stopwatch.StartNew();

**Review by MattMcL4475 — APPROVED** [review]
